streaming/base/storage/download.py

```diff
@@ -314,6 +314,24 @@ def download_from_azure_datalake(remote: str, local: str) -> None:
         raise
 
 
+def download_from_dbfs(remote: str, local: str) -> None:
+    """Download a file from remote Databricks File System to local.
+
+    Args:
+        remote (str): Remote path (dbfs).
+        local (str): Local path (local filesystem).
+    """
+    from databricks.sdk import WorkspaceClient
+
+    client = WorkspaceClient()
+    file_path = remote.lstrip('dbfs:')
+    dbfs_file = client.dbfs.download(file_path)
```

Comment:
is it possible to use `dbutils` to download the file ? For example, `dbutils.fs.cp('source_filepath', 'destination_file_path')` ? The reason for asking is that it might be downloading the shard file faster by splitting the large files into chunks and then downloading those chunks in parallel !!

Reply:
For downloading, I think we should assume for now that DBFS is not mounted on the Mosaic worker filesystem? In that case we will not be able to use dbutils.fs.cp I don't think. But in the case of uploading, if we assume that the user is in a Databricks environment, we can assume the path is mounted. Wdyt?

Reply:
Thanks for the explanation. In that case, let's not use dbutils for now. I am little worried about reading the data serially in chunks (4096 bytes), would it be slow if the shard file size is 128mb or 256mb or even more ? Boto3 and other python sdk under the hood split the file into chunks and downloads it in parallel I suppose !!

Reply:
Unfortunately it doesn't look like the Workspace Client supports parallel downloading. Instead of reading in chunks, should we just read the whole thing at once?

Reply:
Can you try benchmarking the time it takes to download a file by varying chunk size ?

Reply:
Ran a benchmark and it looks more efficient to download all at once.